src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -195,6 +196,18 @@ var ReactCompositeComponentMixin = {
           Component.displayName || Component.name || 'Component'
         );
       }
+
+      var propsMutated = !shallowEqual(inst.props || {}, publicProps);
```

Comment:
`===` just checks if they are the same reference; it is less expensive than `shallowEqual`.